### PR TITLE
Add flake support

### LINKED ISSUE #9: Nix flake support
I use flakes in my personal system config and for various projects. They generally work quite well, better than normal `default.nix` or `shell.nix`. I am setting up a new project and want to try `terranix` but on first glance it doesn't work with flakes. I get an error when I try to run `nix build`.

Here is my flake:
```nix
{
  inputs = { pkgs.url = "github:nixos/nixpkgs/nixos-21.05"; };
  outputs = { self, pkgs }: {
    defaultPackage.x86_64-linux = with import pkgs { system = "x86_64-linux"; };
      stdenv.mkDerivation {
        name = "example";
        src = self;
        buildInputs = [ terranix terraform jq nix ];
        buildPhase = ''
          terranix | jq '.' > config.tf.json
          terraform init
          terraform plan
        '';
      };
  };
}
```

And here is the error:
```
       last 8 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/a0rqhq7wk3r6i3bhc1qp9119w7bb3245-source
       > source root is source
       > patching sources
       > configuring
       > no configure script, doing nothing
       > building
       > error: creating directory '/nix/var': Permission denied
       For full logs, run 'nix log /nix/store/qaqsd36krcs09hxry30h7qgvk5ivk18f-dailp.drv'.
```

I think it originates from `terranix` using `nix-build`, the older system from before flakes, within a flake. Do you think there's an easy fix to this?

### PR DESCRIPTION
**Pull request by l5r**
Added flake support and provided an example. Fixes #9 (I think)

### CODE REVIEW

**Comment by mrVanDalo** [comment]
Thanks @l5r for your pull request. I'll review it once I have time to read myself into flakes. As I wrote in #9 I will do this somewhat this year. (sorry that I did not had time to read into flakes yet).

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      buildTerranix = { pkgs, terranix_config, ... }@terranix_args: let
+        terraform = import ./core/default.nix terranix_args;
+        config_json = pkgs.writeTextFile {
+          name = "terraform-config";
+          text = builtins.toJSON terraform.config;
+          executable = false;
+          destination = "/config.tf.json";
+        };
+      in
+        config_json;
```
Comment (loafofpiecrust) [inline flake.nix:18]: Hi! I am a fan of this project and [use it with flakes](https://github.com/NEU-DSG/dailp-encoding/blob/main/flake.nix#L80). This looks really similar to how I use it and the tips in #9, I might just collapse the `config_json` variable into the body expression here. I'd love to see this merged to bring terranix into the flaky future!

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
```
Comment (loafofpiecrust) [inline flake.nix:4]: I notice that you don't actually use the `flake-utils` input, so I'd take that out.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  outputs = { self, nixpkgs, flake-utils }: rec {
```
Comment (loafofpiecrust) [inline flake.nix:7]: It looks to me like you don't need the `rec` here, and several folks [recommend avoiding it](https://nix.dev/anti-patterns/language#rec-expression) when possible.

File: flake.nix (hunk 1 above)
Comment (mrVanDalo) [reply flake.nix:18]: Hey @loafofpiecrust nice to hear. Sadly I haven't had the time to read into flakes so far. But I will definitely put flakes into terranix in 2021.

**Pull request merged**